roles/ceph-osd/tasks/osd_fragment.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: get osd path
+  shell: "df | grep {{ item }} | awk '{print $6}'"
+  with_items: devices
+  changed_when: false
+  failed_when: false
+  register: osd_path
+
+- name: get osd id
+  command: cat {{ item.stdout }}/whoami
+  with_items: osd_path.results
+  changed_when: false
+  failed_when: false
+  register: osd_id
+
+- name: create a ceph fragment and assemble directory
+  file: path={{ item }}
+        state=directory
+        owner=root
+        group=root
+        mode=0644
+  with_items:
+    - /etc/ceph/ceph.d/
+    - /etc/ceph/ceph.d/osd_fragments
```

Comment:
Pretty sure you can use the file module and it will create leading directories for you, with the same settings as the child directory.